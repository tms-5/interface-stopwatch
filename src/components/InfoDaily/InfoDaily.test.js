import { describe, it, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import InfoDaily from './InfoDaily.vue'

describe('InfoDaily.vue', () => {
    it('renders current date in pt-BR format', () => {
        // Mock Date to ensure consistent test result
        const mockDate = new Date('2023-06-15T12:00:00Z')
        global.Date = class extends Date {
            constructor() {
                super()
                return mockDate
            }
        }
        const wrapper = shallowMount(InfoDaily, {
            props: { timeLeft: 3600, currentTime: '10:00' }
        })
        expect(wrapper.text()).toMatch(/Hoje Ã©.*15 de junho de 2023/)
        // Restore Date
        global.Date = Date
    })

    it('renders current time', () => {
        const wrapper = shallowMount(InfoDaily, {
            props: { timeLeft: 3600, currentTime: '10:00' }
        })
        expect(wrapper.text()).toContain('10:00')
    })

    it('renders end time based on timeLeft', () => {
        const wrapper = shallowMount(InfoDaily, {
            props: { timeLeft: 3600, currentTime: '10:00' }
        })
        // Should render a time string (HH:MM)
        expect(wrapper.text()).toMatch(/\d{2}:\d{2}/)
    })

    it('renders formatted timeLeft', () => {
        const wrapper = shallowMount(InfoDaily, {
            props: { timeLeft: 125, currentTime: '10:00' }
        })
        // 2 minutes and 5 seconds
        expect(wrapper.text()).toContain('02:05')
    })

    it('formats time correctly for 0 seconds', () => {
        const wrapper = shallowMount(InfoDaily, {
            props: { timeLeft: 0, currentTime: '10:00' }
        })
        expect(wrapper.vm.formatTime(0)).toBe('00:00')
    })

    it('formats time correctly for negative or NaN', () => {
        const wrapper = shallowMount(InfoDaily, {
            props: { timeLeft: -10, currentTime: '10:00' }
        })
        expect(wrapper.vm.formatTime(-10)).toBe('00:00')
        expect(wrapper.vm.formatTime(NaN)).toBe('00:00')
    })

    it('formats time correctly for single digit seconds', () => {
        const wrapper = shallowMount(InfoDaily, {
            props: { timeLeft: 8, currentTime: '10:00' }
        })
        expect(wrapper.vm.formatTime(8)).toBe('00:08')
    })

    it('matches snapshot', () => {
        const wrapper = shallowMount(InfoDaily, {
            props: { timeLeft: 90, currentTime: '09:30' }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })
})