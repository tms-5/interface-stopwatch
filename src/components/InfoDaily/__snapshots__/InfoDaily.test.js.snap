// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoDaily.vue > matches snapshot 1`] = `
"<div class="col w-25 g-2 card justify-center">
  <div class="card d-grid" id="daily-date">
    <div class="d-flex g-1 text-start mb-1 align-center">
      <div class="col"><svg width="24" height="25" viewBox="0 0 24 25" fill="none" xmlns="http://www.w3.org/2000/svg">
          <mask id="mask0_3309_287" style="mask-type:alpha;" maskUnits="userSpaceOnUse" x="0" y="0" width="24" height="25">
            <rect y="0.5" width="24" height="24" fill="#D9D9D9"></rect>
          </mask>
          <g mask="url(#mask0_3309_287)">
            <path d="M5 22.5C4.45 22.5 3.97917 22.3042 3.5875 21.9125C3.19583 21.5208 3 21.05 3 20.5V6.5C3 5.95 3.19583 5.47917 3.5875 5.0875C3.97917 4.69583 4.45 4.5 5 4.5H6V2.5H8V4.5H16V2.5H18V4.5H19C19.55 4.5 20.0208 4.69583 20.4125 5.0875C20.8042 5.47917 21 5.95 21 6.5V20.5C21 21.05 20.8042 21.5208 20.4125 21.9125C20.0208 22.3042 19.55 22.5 19 22.5H5ZM5 20.5H19V10.5H5V20.5ZM5 8.5H19V6.5H5V8.5ZM12 14.5C11.7167 14.5 11.4792 14.4042 11.2875 14.2125C11.0958 14.0208 11 13.7833 11 13.5C11 13.2167 11.0958 12.9792 11.2875 12.7875C11.4792 12.5958 11.7167 12.5 12 12.5C12.2833 12.5 12.5208 12.5958 12.7125 12.7875C12.9042 12.9792 13 13.2167 13 13.5C13 13.7833 12.9042 14.0208 12.7125 14.2125C12.5208 14.4042 12.2833 14.5 12 14.5ZM8 14.5C7.71667 14.5 7.47917 14.4042 7.2875 14.2125C7.09583 14.0208 7 13.7833 7 13.5C7 13.2167 7.09583 12.9792 7.2875 12.7875C7.47917 12.5958 7.71667 12.5 8 12.5C8.28333 12.5 8.52083 12.5958 8.7125 12.7875C8.90417 12.9792 9 13.2167 9 13.5C9 13.7833 8.90417 14.0208 8.7125 14.2125C8.52083 14.4042 8.28333 14.5 8 14.5ZM16 14.5C15.7167 14.5 15.4792 14.4042 15.2875 14.2125C15.0958 14.0208 15 13.7833 15 13.5C15 13.2167 15.0958 12.9792 15.2875 12.7875C15.4792 12.5958 15.7167 12.5 16 12.5C16.2833 12.5 16.5208 12.5958 16.7125 12.7875C16.9042 12.9792 17 13.2167 17 13.5C17 13.7833 16.9042 14.0208 16.7125 14.2125C16.5208 14.4042 16.2833 14.5 16 14.5ZM12 18.5C11.7167 18.5 11.4792 18.4042 11.2875 18.2125C11.0958 18.0208 11 17.7833 11 17.5C11 17.2167 11.0958 16.9792 11.2875 16.7875C11.4792 16.5958 11.7167 16.5 12 16.5C12.2833 16.5 12.5208 16.5958 12.7125 16.7875C12.9042 16.9792 13 17.2167 13 17.5C13 17.7833 12.9042 18.0208 12.7125 18.2125C12.5208 18.4042 12.2833 18.5 12 18.5ZM8 18.5C7.71667 18.5 7.47917 18.4042 7.2875 18.2125C7.09583 18.0208 7 17.7833 7 17.5C7 17.2167 7.09583 16.9792 7.2875 16.7875C7.47917 16.5958 7.71667 16.5 8 16.5C8.28333 16.5 8.52083 16.5958 8.7125 16.7875C8.90417 16.9792 9 17.2167 9 17.5C9 17.7833 8.90417 18.0208 8.7125 18.2125C8.52083 18.4042 8.28333 18.5 8 18.5ZM16 18.5C15.7167 18.5 15.4792 18.4042 15.2875 18.2125C15.0958 18.0208 15 17.7833 15 17.5C15 17.2167 15.0958 16.9792 15.2875 16.7875C15.4792 16.5958 15.7167 16.5 16 16.5C16.2833 16.5 16.5208 16.5958 16.7125 16.7875C16.9042 16.9792 17 17.2167 17 17.5C17 17.7833 16.9042 18.0208 16.7125 18.2125C16.5208 18.4042 16.2833 18.5 16 18.5Z" fill="#1C1B1F"></path>
          </g>
        </svg></div>
      <div class="col"><span class="f-1 fw-500">Hoje é</span></div>
    </div><span class="f-1-5 fw-600">qui., 15 de junho de 2023</span>
  </div>
  <div class="card" id="daily-info">
    <div class="d-flex g-1 text-start align-center f-1 fw-600"><svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
        <mask id="mask0_3296_566" style="mask-type:alpha;" maskUnits="userSpaceOnUse" x="0" y="0" width="24" height="24">
          <rect width="24" height="24" fill="#D9D9D9"></rect>
        </mask>
        <g mask="url(#mask0_3296_566)">
          <path d="M5.55 19L2 15.45L3.4 14.05L5.525 16.175L9.775 11.925L11.175 13.35L5.55 19ZM5.55 11L2 7.44999L3.4 6.04999L5.525 8.17499L9.775 3.92499L11.175 5.34999L5.55 11ZM13 17V15H22V17H13ZM13 8.99999V6.99999H22V8.99999H13Z" fill="#1C1B1F"></path>
        </g>
      </svg> Nos atualize: </div>
    <ul class="text-start">
      <li>O que fiz ontem?</li>
      <li>O que farei hoje?</li>
      <li>Algum impedimento?</li>
    </ul>
  </div>
  <div class="card" id="daily-time">
    <div class="d-flex g-1 text-start mb-1 align-center">
      <div class="col"><svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
          <mask id="mask0_3296_574" style="mask-type:alpha;" maskUnits="userSpaceOnUse" x="0" y="0" width="24" height="24">
            <rect width="24" height="24" fill="#D9D9D9"></rect>
          </mask>
          <g mask="url(#mask0_3296_574)">
            <path d="M11.9996 22C10.7496 22 9.57878 21.7625 8.48711 21.2875C7.39544 20.8125 6.44544 20.1708 5.63711 19.3625C4.82878 18.5542 4.18711 17.6042 3.71211 16.5125C3.23711 15.4208 2.99961 14.25 2.99961 13C2.99961 11.75 3.23711 10.5792 3.71211 9.48751C4.18711 8.39584 4.82878 7.44584 5.63711 6.63751C6.44544 5.82917 7.39544 5.18751 8.48711 4.71251C9.57878 4.23751 10.7496 4.00001 11.9996 4.00001C13.2496 4.00001 14.4204 4.23751 15.5121 4.71251C16.6038 5.18751 17.5538 5.82917 18.3621 6.63751C19.1704 7.44584 19.8121 8.39584 20.2871 9.48751C20.7621 10.5792 20.9996 11.75 20.9996 13C20.9996 14.25 20.7621 15.4208 20.2871 16.5125C19.8121 17.6042 19.1704 18.5542 18.3621 19.3625C17.5538 20.1708 16.6038 20.8125 15.5121 21.2875C14.4204 21.7625 13.2496 22 11.9996 22ZM14.7996 17.2L16.1996 15.8L12.9996 12.6V8.00001H10.9996V13.4L14.7996 17.2ZM5.59961 2.35001L6.99961 3.75001L2.74961 8.00001L1.34961 6.60001L5.59961 2.35001ZM18.3996 2.35001L22.6496 6.60001L21.2496 8.00001L16.9996 3.75001L18.3996 2.35001ZM11.9996 20C13.9496 20 15.6038 19.3208 16.9621 17.9625C18.3204 16.6042 18.9996 14.95 18.9996 13C18.9996 11.05 18.3204 9.39584 16.9621 8.03751C15.6038 6.67917 13.9496 6.00001 11.9996 6.00001C10.0496 6.00001 8.39544 6.67917 7.03711 8.03751C5.67878 9.39584 4.99961 11.05 4.99961 13C4.99961 14.95 5.67878 16.6042 7.03711 17.9625C8.39544 19.3208 10.0496 20 11.9996 20Z" fill="#1C1B1F"></path>
          </g>
        </svg></div>
      <div class="col align-center"><span class="f-1 fw-500">Hora agora</span></div>
    </div><span class="f-2 fw-600">09:30</span>
  </div>
  <div class="card" id="daily-time-end">
    <div class="d-flex g-1 text-start mb-1 align-center">
      <div class="col"><svg width="24" height="25" viewBox="0 0 24 25" fill="none" xmlns="http://www.w3.org/2000/svg">
          <mask id="mask0_3310_293" style="mask-type:alpha;" maskUnits="userSpaceOnUse" x="0" y="0" width="24" height="25">
            <rect y="0.5" width="24" height="24" fill="#D9D9D9"></rect>
          </mask>
          <g mask="url(#mask0_3310_293)">
            <path d="M10.025 24.5C8.64167 24.5 7.34167 24.2375 6.125 23.7125C4.90833 23.1875 3.84583 22.4708 2.9375 21.5625C2.02917 20.6542 1.3125 19.5917 0.7875 18.375C0.2625 17.1583 0 15.8667 0 14.5C0 13.1167 0.2625 11.8167 0.7875 10.6C1.3125 9.38334 2.02917 8.32084 2.9375 7.4125C3.84583 6.50417 4.90833 5.7875 6.125 5.2625C7.34167 4.7375 8.64167 4.475 10.025 4.475C10.1917 4.475 10.3542 4.47917 10.5125 4.4875C10.6708 4.49584 10.8333 4.50834 11 4.525V6.525C10.8333 6.50834 10.6708 6.49584 10.5125 6.4875C10.3542 6.47917 10.1917 6.475 10.025 6.475C7.79167 6.475 5.89583 7.25417 4.3375 8.8125C2.77917 10.3708 2 12.2667 2 14.5C2 16.7167 2.77917 18.6042 4.3375 20.1625C5.89583 21.7208 7.79167 22.5 10.025 22.5C12.2417 22.5 14.1292 21.7208 15.6875 20.1625C17.2458 18.6042 18.025 16.7167 18.025 14.5H20.025C20.025 15.8667 19.7625 17.1583 19.2375 18.375C18.7125 19.5917 17.9958 20.6542 17.0875 21.5625C16.1792 22.4708 15.1208 23.1875 13.9125 23.7125C12.7042 24.2375 11.4083 24.5 10.025 24.5ZM13.3 19.2L9 14.9V9.5H11V14.1L14.7 17.8L13.3 19.2ZM21.85 12.5C21.4667 10.1333 20.4667 8.07917 18.85 6.3375C17.2333 4.59584 15.2833 3.43334 13 2.85V0.800003C15.85 1.41667 18.2708 2.79167 20.2625 4.925C22.2542 7.05834 23.4583 9.58334 23.875 12.5H21.85ZM17.775 12.5C17.4583 11.25 16.8792 10.15 16.0375 9.2C15.1958 8.25 14.1833 7.53334 13 7.05V4.925C14.7167 5.45834 16.1917 6.40417 17.425 7.7625C18.6583 9.12084 19.4583 10.7 19.825 12.5H17.775Z" fill="#1C1B1F"></path>
          </g>
        </svg></div>
      <div class="col"><span class="f-1 fw-500">Horário que a daily acaba</span></div>
    </div><span class="f-2 fw-600">09:00</span>
  </div>
  <div class="card" id="daily-time-left">
    <div class="d-flex g-1 text-start mb-1 align-center">
      <div class="col"><svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
          <mask id="mask0_3296_557" style="mask-type:alpha;" maskUnits="userSpaceOnUse" x="0" y="0" width="24" height="24">
            <rect width="24" height="24" fill="#D9D9D9"></rect>
          </mask>
          <g mask="url(#mask0_3296_557)">
            <path d="M8 20H16V17C16 15.9 15.6083 14.9583 14.825 14.175C14.0417 13.3917 13.1 13 12 13C10.9 13 9.95833 13.3917 9.175 14.175C8.39167 14.9583 8 15.9 8 17V20ZM4 22V20H6V17C6 15.9833 6.2375 15.0292 6.7125 14.1375C7.1875 13.2458 7.85 12.5333 8.7 12C7.85 11.4667 7.1875 10.7542 6.7125 9.8625C6.2375 8.97083 6 8.01667 6 7V4H4V2H20V4H18V7C18 8.01667 17.7625 8.97083 17.2875 9.8625C16.8125 10.7542 16.15 11.4667 15.3 12C16.15 12.5333 16.8125 13.2458 17.2875 14.1375C17.7625 15.0292 18 15.9833 18 17V20H20V22H4Z" fill="#1C1B1F"></path>
          </g>
        </svg></div>
      <div class="col"><span class="f-1 fw-500">Tempo restante para acabar a daily</span></div>
    </div><span class="f-2 fw-600">01:30</span>
  </div>
</div>"
`;
